### ISSUE TITLE
netlify env:set - TypeError: Cannot read properties of undefined (reading 'env')

### ISSUE THREAD

**Issue by chrisrabe** [netlify/cli#5417]
### Describe the bug

Using `netlify env:set or env:import` throws the following error:

```
›   TypeError: Cannot read properties of undefined (reading 'env')
```

### Steps to reproduce

1. [Deploy NextJS app](https://www.netlify.com/blog/2020/11/30/how-to-deploy-next.js-sites-to-netlify/)
2. Add environment variable `netlify env:set BANANA banana`
3. See it throw error

![Screenshot 2023-01-22 at 6 02 45 pm](https://user-images.githubusercontent.com/11940900/213906328-15314339-1ae1-40ec-a588-eabf4f20463d.png)

### Configuration

```
[build]
  command = "npm run build"
  functions = "netlify/functions"
  publish = ".next"

  ## Uncomment to use this redirect for Single Page Applications like create-react-app.
  ## Not needed for static site generators.
  #[[redirects]]
  #  from = "/*"
  #  to = "/index.html"
  #  status = 200

  ## (optional) Settings for Netlify Dev
  ## https://github.com/netlify/cli/blob/main/docs/netlify-dev.md#project-detection
  #[dev]
  #  command = "yarn start" # Command to start your dev server
  #  port = 3000 # Port that the dev server will be listening on
  #  publish = "dist" # Folder with the static content for _redirect file

  ## more info on configuring this file: https://docs.netlify.com/configure-builds/file-based-configuration/

[context.production]
  NODE_ENV = "production"
  UI_URL = "https://uat-loopin.netlify.app"
  NEW_RELIC_APP_NAME = "loopin-ui_PROD"
  NEW_RELIC_ENABLED = "true"
  ANALYTICS_COLLECTION_ENABLED = "true"

[context.deploy-preview.environment]
  NODE_ENV = "development"
  UI_URL = "https://preview--uat-loopin.netlify.app"
  NEW_RELIC_APP_NAME = "loopin-ui_DEV"
  NEW_RELIC_ENABLED = "true"
  ANALYTICS_COLLECTION_ENABLED = "true"

[context.branch-deploy.environment]
  NODE_ENV = "development"
  UI_URL = "https://preview--uat-loopin.netlify.app"
  NEW_RELIC_APP_NAME = "loopin-ui_DEV"
  NEW_RELIC_ENABLED = "true"
  ANALYTICS_COLLECTION_ENABLED = "true"

[context.staging.environment]
  NODE_ENV = "staging"
  UI_URL = "https://preview--uat-loopin.netlify.app"
  NEW_RELIC_APP_NAME = "loopin-ui_STG"
  NEW_RELIC_ENABLED = "true"
  ANALYTICS_COLLECTION_ENABLED = "true"
```

### Environment

```
  System:
    OS: macOS 13.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 63.55 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
    Watchman: 2022.02.07.00 - /usr/local/bin/watchman
  npmPackages:
    netlify-cli: ^12.9.1 => 12.9.1 
```

**Comment by tinfoil-knight** [comment]
Hey @chrisrabe,
I followed the steps in the linked blog post, copied your configuration & then tried the `netlify env:set` command. I was able to successfully set the environment variables & didn't receive any error.

Could you share a public repository where you're able to reproduce the issue so that I can try to inspect & fix it?

**Comment by chrisrabe** [comment]
Unfortunately this is from a private repository. Would having the repo in a private repository cause this problem?

**Comment by tinfoil-knight** [comment]
> Unfortunately this is from a private repository. Would having a private repository cause this problem?

Nope. Having a private repository won't cause this issue.

**Comment by chrisrabe** [comment]
After 1 month, I finally figured out the issue. It turns out that the `siteId` property of the `state.json` file inside the `.netlify` directory was incorrect in my repository. 

I accidentally found the issue when I tried to run `netlify env:list` and it returned a message saying "Cannot find environment variables of site: undefined".

I fixed it by:

1. Connecting Netlify to the Bitbucket repository and deploying the app through the Netlify dashboard
2. Running `netlify sites:list` to get the proper `siteId`
3. Replacing the "siteId" inside `state.json` with the correct siteId

After that `netlify deploy` and `netlify env:list` and other commands are now working. Can happily close this issue now.
